### ISSUE TITLE
Can't Install Appium on Windows

### ISSUE THREAD

**Issue by Grimlek** [appium/appium#6390]
## The problem

I can't seem to install Appium on windows using npm. I tried re-installation of node and setting permissions to the local folder.

C:\Users\user\AppData\Local

## Environment

```
npm install -g appium


npm WARN peerDependencies The peer dependency continuation-local-storage@~3 incl
uded from cls-bluebird will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerD
ependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it
explicitly.
npm WARN peerDependencies The peer dependency bluebird@>=1.0.3 included from cls
-bluebird will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerD
ependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it
explicitly.

> appium-chromedriver@2.8.3 install C:\User\charles.sexton\AppData\Local\node_mo
dules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver

> node install-npm.js

Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
Error trying to install Chromedriver binary. Waiting and trying again. Cannot fi
nd module '../modules/web.dom.iterable'
info Chromedriver Install Installing Chromedriver version '2.21' for platform 'w
in' and architecture '32'
info Chromedriver Install Opening temp file to write chromedriver_win32 to...
info Chromedriver Install Downloading http://chromedriver.storage.googleapis.com
/2.21/chromedriver_win32.zip...
info Chromedriver Install Writing binary content to C:\Users\CHARLE~1.SEX\AppDat
a\Local\Temp\2016311-4848-1s1ggk0\chromedriver_win32.zip...
info Chromedriver Install Extracting C:\Users\CHARLE~1.SEX\AppData\Local\Temp\20
16311-4848-1s1ggk0\chromedriver_win32.zip to C:\Users\CHARLE~1.SEX\AppData\Local
\Temp\2016311-4848-1s1ggk0\chromedriver_win32
info Chromedriver Install Creating C:\User\charles.sexton\AppData\Local\node_mod
ules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\
chromedriver\win...
info Chromedriver Install Copying unzipped binary, reading from C:\Users\CHARLE~
1.SEX\AppData\Local\Temp\2016311-4848-1s1ggk0\chromedriver_win32\chromedriver.ex
e...
info Chromedriver Install Writing to C:\User\charles.sexton\AppData\Local\node_m
odules\appium\node_modules\appium-android-driver\node_modules\appium-chromedrive
r\chromedriver\win\chromedriver.exe...
info Chromedriver Install C:\User\usr\AppData\Local\node_modules\appi
um\node_modules\appium-android-driver\node_modules\appium-chromedriver\chromedri
ver\win\chromedriver.exe successfully put in place

> appium-selendroid-driver@1.3.1 install C:\User\usr\AppData\Local\no
de_modules\appium\node_modules\appium-selendroid-driver
> node ./bin/install.js

dbug AndroidDriver Getting Java version
info SelendroidInstaller Ensuring C:\User\usr\AppData\Local\node_modu
les\appium\node_modules\appium-selendroid-driver\node_modules\appium-selendroid-
installer\selendroid\download exists
Error: Command 'java -version' exited with code 2
    at ChildProcess.<anonymous> (lib/teen_process.js:56:19)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:827:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:477:12)
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x8
6)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "appium"
npm ERR! node v4.4.2
npm ERR! npm  v2.15.0
npm ERR! code ELIFECYCLE

npm ERR! appium-selendroid-driver@1.3.1 install: `node ./bin/install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the appium-selendroid-driver@1.3.1 install script 'node ./bin
/install.js'.
npm ERR! This is most likely a problem with the appium-selendroid-driver package
,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./bin/install.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs appium-selendroid-driver
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls appium-selendroid-driver
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\usr\npm-debug.log
```

**Comment by scottdixon** [comment]
Do you have java installed & in your path?

**Comment by Grimlek** [comment]
I have jdk 1.8 as the JAVA_HOME var and referenced the java home var in Path var.

**Comment by Grimlek** [comment]
info AppiumDoctor ### Diagnostic starting ###
info AppiumDoctor  ? ANDROID_HOME is set to: C:\Users\charles.sexton\AppData\Loc
al\Android\sdk\
info AppiumDoctor  ? JAVA_HOME is set to: C:\Program Files\Java\jdk1.8.0_73
info AppiumDoctor  ? adb exists at: C:\Users\charles.sexton\AppData\Local\Androi
d\sdk\platform-tools\adb.exe
info AppiumDoctor  ? android exists at: C:\Users\charles.sexton\AppData\Local\An
droid\sdk\tools\android.bat
info AppiumDoctor  ? emulator exists at: C:\Users\charles.sexton\AppData\Local\A
ndroid\sdk\tools\emulator.exe
info AppiumDoctor ### Diagnostic completed, no fix needed. ###
info AppiumDoctor
info AppiumDoctor Everything looks good, bye!
info AppiumDoctor

**Comment by imurchie** [comment]
What is the output of the `java -version` command?

We might need to make our checks different for different systems.

**Comment by Grimlek** [comment]
I changed to 1.7 from 1.8 to see if that would make a difference and the same error occurred. Just wanted to explain why the the java -version command showed 1.7 instead of 1.8.

java version "1.7.0_80"
Java(TM) SE Runtime Environment (build 1.7.0_80-b15)
Java HotSpot(TM) 64-Bit Server VM (build 24.80-b11, mixed mode)

**Comment by Grimlek** [comment]
I also tried issue [#6329](https://github.com/appium/appium/issues/6329) but produced the same error

**Comment by imurchie** [comment]
Ah! I'm glad you got it running. Were you using our documentation to install, and this step was confusing? If so, I'll update the documentation.

**Comment by Grimlek** [comment]
I looked at your documentation while troubleshooting the above error, it wasn't of any help. Also while looking at the documentation you might to include that Window users need to have a path variable setup for appium-doctor pointing to the directory of appium-doctor.cmd. I had an issue trying to run the appium-doctor command until I added a variable which directly points to the directory location of this command file. I would also update the documentation for the above problem as I don't see it in the documentation very clearly.

**Comment by imurchie** [comment]
Thank you for the feedback. Did it not work to install `npm install appium-doctor -g` and then run `appium-doctor`?

**Comment by Grimlek** [comment]
No sir it didn't. As stated above I had to include in the windows Path variable the location of the appium-connect.cmd file. It installed appium-doctor just fine and I could use node and npm window command but I couldn't use appium-doctor after installation of appium-doctor in the global node modules folder. I had to provide the directory path in the windows system variable Path pointing to appium-doctor-cmd file.
